tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnectionTests.cs

```diff
@@ -276,6 +276,46 @@ public void Dispose_should_raise_the_correct_events()
             act.ShouldThrow<InvalidOperationException>();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void ReceiveMessage_should_throw_error_when_message_is_an_invalid_size(
+            [Values(-1, 16000005)]
```

Comment:
I was expecting 48000001 as the second test value (48000000 + 1).

In order to use that value let's change the `BinaryConnectionTests` constructor to use some more realistic setups.

Lines 54-67 of the constructor could be refactored to have more realistic values for isMasterResult and buildInfoResult:

    _endPoint = new DnsEndPoint("localhost", 27017);
    var serverId = new ServerId(new ClusterId(), _endPoint);
    var connectionId = new ConnectionId(serverId);
    var isMasterResult = new IsMasterResult(new BsonDocument { { "ok", 1 }, { "maxMessageSizeBytes", 48000000 } });
    var buildInfoResult = new BuildInfoResult(new BsonDocument { { "ok", 1 }, { "version", "2.6.3" } });
    var connectionDescription = new ConnectionDescription(connectionId, isMasterResult, buildInfoResult);

    _mockConnectionInitializer = new Mock<IConnectionInitializer>();
    _mockConnectionInitializer
        .Setup(i => i.InitializeConnection(It.IsAny<IConnection>(), CancellationToken.None))
        .Returns(connectionDescription);
    _mockConnectionInitializer
        .Setup(i => i.InitializeConnectionAsync(It.IsAny<IConnection>(), CancellationToken.None))
        .Returns(Task.FromResult(connectionDescription));

Reply:
Should this be `(48*1024*1024*) + 1` (per the discussion about 48MB vs 48000)?

Reply:
No. Oddly enough the MaxMessageSize is 48,000,000, not 48MB.